### PR TITLE
fix: Fork button text color/hover, and fork=true query param fix

### LINKED ISSUE #26295: [Bug]: App navigation fork button issues
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Fund 2 issues with app navigation. 
1. ForkApp button text at the top right is not visible until hovered.
    On the dev tools button text has color set to :rgb(75, 72, 72) !important. This works if apps theme is light colored but not in above app's case.
2. ~If we pass `?showForkTemplateModal=true`  modal is not coming up.
Once landed on page, click on Fork app it comes up.~ [No longer an issue, see the comment below.](https://github.com/appsmithorg/appsmith/issues/26295#issuecomment-1723148678)
3. The fork modal doesn't close when we set fork=true. https://lcixo7xrxt.vmaker.com/record/TYYVeNm9BucsJRNa

### Steps To Reproduce

Can be tested in any publicly accessible app. 

### Public Sample App

https://app.appsmith.com/app/websocket-poc-with-crypto/websocket-with-mutations-647f8d690cd629545d130b68?showForkTemplateModal=true

### Environment

Production

### Issue video log

_No response_

### Version

Appsmith v1.9.32-SNAPSHOT

### PR DESCRIPTION
**Pull request by dhruvikn**
#### PR fixes following issue(s)
Fixes #26295

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by dhruvikn** [comment]
Whoever picks this up for QA, please note that we'll need to test all app forking scenarios. However, adding the query param `fork=true` should only work when appended in the URL in deployed/app viewer mode.

cc: @laveena-en @chandannkumar

**Review by rahulbarwal — APPROVED** [review]
Looks good. One question though: This `fork` param should have been covered by a cypress spec. Is it not there? if not, we should add with this PR to make sure we don't get the bug you found yesterday - for both cancel and fork buttons on modal.

**Comment by chandannkumar** [comment]
Tested this PR and LGTM

File: app/client/src/pages/AppViewer/PrimaryCTA.tsx [hunk 1]
```diff
@@ -70,6 +70,48 @@ function PrimaryCTA(props: Props) {
...
+
+  useEffect(() => {
+    if (queryParams.get("fork") === "true") {
+      handleForkModalOpen();
+    } else {
+      handleForkModalClose();
+    }
+  }, []);
+
+  const appendOrDeleteForkParam = (appendOrDelete: "apend" | "delete") => {
```
Comment (akash-codemonk) [inline app/client/src/pages/AppViewer/PrimaryCTA.tsx:84]: nit: append

**Review by rahulbarwal — APPROVED** [review]
Looks good. 
Curios question: I don't see any usage of `<WidgetSidebar/>` Why do we have this in this PR then?

**Comment by dhruvikn** [comment]
> Looks good. Curios question: I don't see any usage of `<WidgetSidebar/>` Why do we have this in this PR then?

@rahulbarwal, I recently merged [a PR](https://github.com/appsmithorg/appsmith/pull/27401) that removes widgetsidebar. There must be some mess up on my local git, I'll fix it. Thanks for the sharp attention!

**Comment by dhruvikn** [comment]
@rahulbarwal - could you please add your review again? Have removed the WidgetSidebar.tsx

**Review by rahulbarwal — APPROVED** [review]

**Comment by dhruvikn** [comment]
Adding "Test Plan Approved" label [wrt Chandan's approval here](https://github.com/appsmithorg/appsmith/pull/27336#issuecomment-1727172858).

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**